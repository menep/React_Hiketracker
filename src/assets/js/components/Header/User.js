import React from "react";
import { NavLink } from "react-router-dom";

const User = props => (
  <ul className="user__container">
    {/* <li className="user__auth-btn">
    </li>
    <li className="user__auth-btn">Sign Up</li> */}
    <li className="user__auth-btn">
      <NavLink
        to="/my-hikes"
        className="user__auth-btn--inactive"
        activeClassName="user__auth-btn--active"
      >
        My Hikes
      </NavLink>
    </li>
  </ul>
);

export default User;
