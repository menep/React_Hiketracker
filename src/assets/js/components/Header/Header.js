import React from "react";
import { NavLink } from "react-router-dom";
import User from "./User";

const Header = () => (
  <header className="header__container">
    <div className="header__logo">
      <h2>
        <NavLink exact
          to="/"
          className="header__heading"
          activeClassName="header__heading--active"
        >
          HTKR
        </NavLink>
      </h2>
    </div>
    <User />
  </header>
);

export default Header;
