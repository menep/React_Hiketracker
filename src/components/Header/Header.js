import React from "react";
import User from "../User/User";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header className="header__container">
    <div className="header__logo">
      <h2>
        <NavLink exact
          to="/"
          className="header__heading"
          activeClassName="header__heading--active"
        >
          HTKR
        </NavLink>
      </h2>
    </div>
    <User />
  </header>
);

export default Header;
