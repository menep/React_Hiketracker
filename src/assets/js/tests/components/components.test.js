import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Header from "../../components/Header/Header";
import User from "../../components/Header/User";

test("User renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <User />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Header renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
