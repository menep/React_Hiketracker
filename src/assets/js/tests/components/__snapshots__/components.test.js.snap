// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header__container"
>
  <div
    className="header__logo"
  >
    <h2>
      <a
        aria-current="page"
        className="header__heading header__heading--active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        HTKR
      </a>
    </h2>
  </div>
  <ul
    className="user__container"
  >
    <li
      className="user__auth-btn"
    >
      <a
        aria-current={null}
        className="user__auth-btn--inactive"
        href="/my-hikes"
        onClick={[Function]}
      >
        My Hikes
      </a>
    </li>
  </ul>
</header>
`;

exports[`User renders correctly 1`] = `
<ul
  className="user__container"
>
  <li
    className="user__auth-btn"
  >
    <a
      aria-current={null}
      className="user__auth-btn--inactive"
      href="/my-hikes"
      onClick={[Function]}
    >
      My Hikes
    </a>
  </li>
</ul>
`;
